client/apps/edit/components/section_image_collection/index.coffee

```diff
@@ -115,20 +116,19 @@ module.exports = React.createClass
     e.preventDefault()
     val = @refs.byUrl.value
     slug = _.last(val.split '/')
-    $(@refs.byUrl).siblings('button').addClass('is-loading')
+    @setState isLoading: !@state.isLoading
+    @refs.byUrl.value = ''
```

Comment:
The same pattern would apply here. Instead of setting `@refs.byUrl.value = ''` or `@refs.byUrl.placeholder = ` you would use `@setState` and reflect the updated UI in the `render function.

e.g.

````coffeescript
componentDidMount: ->
  @setState isLoading: !@state.isLoading
  new Artwork().fetch
    error: ->
      @setState placeholder: 'Artwork not found' if res.status if 404
      setTimeout => @setState placeholder: 'Add artwork url', isLoading: false

render: ->
  input {
    value: if @state.isLoading then '' else null
    placeholder: @state.placeholder
  },
````

_The React Way_ is that basically _everything_ that can change over time should go into state and if it's "config" that doesn't change over time it passes through props. Using refs or manipulating elements directly is more of an anti-pattern.

I hope this is helpful and not annoyingly pedantic 😶